ui/apps/platform/src/Containers/Clusters/ClustersTable.tsx

```diff
@@ -0,0 +1,246 @@
+                    <>
+                        {data.map((clusterInfo, rowIndex) => {
+                            const provider = formatCloudProvider(
+                                clusterInfo.status?.providerMetadata
+                            );
+                            const clusterId = clusterInfo.id;
+                            return (
+                                <Tbody isExpanded={isRowExpanded(clusterId)} key={clusterInfo.id}>
+                                    <Tr key={clusterInfo.id}>
+                                        <Td
+                                            select={{
+                                                rowIndex,
+                                                onSelect: () => toggleCluster(clusterId),
+                                                isSelected: selectedClusterIds.includes(clusterId),
+                                            }}
+                                        />
+                                        <Td dataLabel="Cluster">
+                                            <Flex
+                                                alignItems={{ default: 'alignItemsCenter' }}
+                                                columnGap={{ default: 'columnGapXs' }}
+                                                flexWrap={{ default: 'nowrap' }}
+                                            >
+                                                <Link to={clusterId} className="">
+                                                    {clusterInfo.name}
+                                                </Link>
+                                                {isHelmManaged(clusterInfo) && <HelmIndicator />}
+                                                {isOperatorManaged(clusterInfo) && (
+                                                    <OperatorIndicator />
+                                                )}
+                                            </Flex>
+                                        </Td>
+                                        <Td dataLabel="Provider (Region)">{provider}</Td>
+                                        <Td
+                                            dataLabel="Status"
+                                            compoundExpand={{
+                                                isExpanded: isCellExpanded(clusterId, COL.STATUS),
+                                                onToggle: () => toggle(clusterId, COL.STATUS),
+                                                rowIndex,
+                                                columnIndex: 3,
+                                            }}
+                                        >
+                                            {/* TODO: needs update for upgrade */}
+                                            <ClusterStatus
+                                                healthStatus={
+                                                    clusterInfo?.healthStatus as ClusterHealthStatus
```

Comment:
In possible follow ups, `ClusterStatus` (and other component) props might be declared better (than I did)  to indicate that it handles absence of value:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Clusters/Components/ClusterStatus.tsx#L30

I think the incomplete declarations were less apparent with indirect rendering via descriptors.

Reply:
I agree 100%. This is just a placeholder, I'll likely be redoing this component entirely for the patternfly upgrade.